tests/Install-Scripts.Test/GivenThatIWantToInstallTheSdkFromAScript.cs

```diff
@@ -0,0 +1,303 @@
+        };
+
+        public static IEnumerable<object?[]> ChannelRuntimeTestCases
+        {
+            get
+            {
+                // Download SDK using branches as channels
+                foreach(var sdkBranchInfo in _sdkBranches)
+                {
+                    yield return new object?[]
+                    {
+                        sdkBranchInfo.branch,
+                        sdkBranchInfo.versionRegex,
+                        null,
+                        null,
+                    };
+                }
+
+                // Download runtimes using branches as channels
+                foreach(var runtimeBranchInfo in _runtimeBranches)
+                {
+                    foreach(var runtime in _runtimes)
+                    {
+                        yield return new object?[]
+                        {
+                            runtimeBranchInfo.branch,
+                            runtimeBranchInfo.versionRegex,
+                            runtime.runtime,
+                            runtime.runtimeIdentifier,
+                        };
+                    }
+                }
+
+                // Download SDK and runtimes from darc channels.
+                foreach(var channelInfo in _channels)
+                {
+                    yield return new object?[]
+                    {
+                        channelInfo.channel,
+                        channelInfo.versionRegex,
+                        null,
+                        null,
+                    };
+
+                    foreach (var runtimeInfoe in _runtimes)
+                    {
+                        yield return new object?[]
+                        {
+                            channelInfo.channel,
+                            channelInfo.versionRegex,
+                            runtimeInfoe.runtime,
+                            runtimeInfoe.runtimeIdentifier,
+                        };
+                    }    
+                }
+            }
+        }
+
+        /// <summary>
+        /// The directory that the install script will install the .NET into.
+        /// </summary>
+        private readonly string _sdkInstallationDirectory;
+
+        /// <summary>
+        /// Instantiates a GivenThatIWantToInstallTheSdkFromAScript instance.
+        /// </summary>
+        /// <remarks>This constructor is called once for each of the tests to run.</remarks>
+        public GivenThatIWantToInstallTheSdkFromAScript()
+        {
+            _sdkInstallationDirectory = Path.Combine(
+                Path.GetTempPath(),
+                "InstallScript-Tests");
+
+            // In case there are any files from previous runs, clean them up.
+            try
+            {
+                Directory.Delete(_sdkInstallationDirectory, true);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                // This is expected. Ignore the exception.
```

Comment:
Why not check the existence of the directory before the deletion? It seems strange to have the normal flow of execution goes through the catch block.

Reply:
If the following events happen in order, we could still get an exception:
- We check if the directory exists and it returns true. We think we are safe to delete the folder.
- Some other process/user deletes the directory (windows maybe, because the system is running out of disk)
- We attempt to delete the folder. We get an exception.
Since we have to check for the exception anyway, I think this makes it unnecessary to check for the existence of the directory.